crates/provider/src/native.rs

```diff
@@ -80,6 +74,15 @@ impl<T: FileSystem + Send + Sync> NativeProvider<T> {
             filesystem,
         }
     }
+
+    pub fn get_node_from_name(&self, node_name: &str) -> Result<&Process, ProviderError> {
```

Comment:
```suggestion
   fn get_process_by_node_name(&self, node_name: &str) -> Result<&Process, ProviderError> {
```
I think this is a better naming, since we are getting a ref of the process (by node name). Also, this method doesn't need to be public since will only be used internally.